app/api/resources/user.py

```diff
@@ -498,3 +498,111 @@ def get(cls):
             return messages.USER_NOT_FOUND, HTTPStatus.NOT_FOUND
 
         return dashboard, HTTPStatus.OK
+
+@users_ns.route("user/forgot_password")
+class ForgotPassword(Resource):
+    @classmethod
+    @users_ns.doc(f"forgot password")
+    @users_ns.response(HTTPStatus.OK, f"{messages.PASSWORD_RESET_MAIL_MESSAGE}")
+    @users_ns.response(HTTPStatus.BAD_REQUEST, f"{messages.EMAIL_FIELD_IS_MISSING}")
+    @users_ns.response(HTTPStatus.FORBIDDEN, f"{messages.USER_HAS_NOT_VERIFIED_EMAIL_BEFORE_LOGIN}")
+    @users_ns.response(HTTPStatus.NOT_FOUND, f"{messages.USER_IS_NOT_REGISTERED_IN_THE_SYSTEM}")
+    @users_ns.expect(forgot_password_change_request_data_model, validate=True)
+ 
+    def post(self):
+        """ Updates User's password by sending reset link to email
+ 
+        Args: 
+          email : Verified email of the user
+ 
+        Return:
+          token: a token link/reset link will be shared on verified mail of user
+        """
+ 
+        data = request.json
+ 
+        if not data:
+            return messages.EMAIL_FIELD_IS_MISSING, HTTPStatus.BAD_REQUEST
+ 
+        is_valid = validate_forgot_password_email(data)
+ 
+        if is_valid!={}:
+            return is_valid, HTTPStatus.BAD_REQUEST
+        
+        user = DAO.get_user_by_email(data["email"])
+ 
+        if user is None:
+            return messages.USER_IS_NOT_REGISTERED_IN_THE_SYSTEM, HTTPStatus.NOT_FOUND
+ 
+        if not user.is_email_verified:
+            return messages.USER_HAS_NOT_VERIFIED_EMAIL_BEFORE_LOGIN, HTTPStatus.FORBIDDEN    
+ 
+        send_email_reset_password_message(user.name, data["email"])
+ 
+        return messages.PASSWORD_RESET_MAIL_MESSAGE, HTTPStatus.OK
+ 
+@users_ns.route("user/reset_password")
+@users_ns.response(HTTPStatus.OK, f"{messages.PASSWORD_SUCCESSFULLY_UPDATED}")
+@users_ns.response(
+    HTTPStatus.UNAUTHORIZED,
+    "%s\n%s\n%s"
```

Comment:
if you could use f-strings here as well ;)

Reply:
will do this soon!

Reply:
Sorry, can't do this as I just saw that backslashes may not appear inside the expression portions of f-strings, [...] as in this article but if you know how to do this then please help me to correct this as i am not much aware about f-string. i did earlier ones as they were easy but this one , i am not able to correct. so, i saw in the entire file , there are other places as well where no f-string was used so, what to do in that case ?

Reply:
Sorry, can't do this as I just saw that backslashes may not appear inside the expression portions of f-strings, [...] as in this article but if you know how to do this then please help me to correct this as i am not much aware about f-string. i did earlier ones as they were easy but this one , i am not able to correct. so, i saw in the entire file , there are other places as well where no f-string was used so, what to do in that case ?

Reply:
Hi @isabelcosta i used \n separately in between f-strings so, i was able to achieve that required formate . Now, can please check this PR again (IF YOU HAVE TIME)